jenkins/automation/splint-memory-leak.groovy

```diff
@@ -0,0 +1,93 @@
+
+pipeline {
+    agent {
+        node {
+            label "docker-centos-7.9.2009-node"
+        }
+    }
+
+    options {
+        timestamps()
+        buildDiscarder(logRotator(daysToKeepStr: '30', numToKeepStr: '30'))
+        ansiColor('xterm')
+    }
+
+    parameters {
+        string(name: 'CORTX_MOTR_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for Memory Leak Analysis', trim: true)
+        string(name: 'CORTX_MOTR_REPO', defaultValue: 'https://github.com/Seagate/cortx-motr/', description: 'Repository for Memory Leak Analysis', trim: true)
+    }
+
+    stages {
+        stage('Checkout Script') {
+            steps {
+                cleanWs()
+                script {
+                    checkout([$class: 'GitSCM', branches: [[name: "${CORTX_MOTR_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: "${CORTX_MOTR_REPO}"]]])
+                }
+            }
+        }
+
+        stage ('Execute Memory Analysis') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: '', returnStatus: true, script: '''
+                yum install splint -y
+                mkdir -p artifacts
+                for file in  $(find $PWD -type f -regextype posix-egrep -regex ".+\\\\\\\\.(c|h|lcl)$"); do splint +trytorecover -preproc -warnposix +line-len 100000 $file | sed \'H;1h;$!d;g;s/\\n  */ /g\' >> artifacts/splint-analysis.log ; done || echo "Log Generated"
```

Comment:
@shailesh-vaidya, can we echo message to `ERROR: splint execution failed. please check logs for more details`

Reply:
Fixed and tested in https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/Motr/job/memory-leak-analysis/19/